packages/infrastructure/src/public/saleor/store/types.ts

```diff
@@ -34,14 +39,28 @@ export type GetProductDetailsUseCase = (opts: ProductDetailOptions) => Promise<
   | { data: null; errors: unknown[] }
 >;
 
-export type StoreService<Config> = (config: Config) => {
-  getProductDetails: GetProductDetailsUseCase;
-};
-
 export type GetProductAvailabilityDetailsInfra = (
   opts: ProductDetailOptions,
 ) => Promise<{ availability: ProductAvailability | null; errors: unknown[] }>;
 
 export type GetProductDetailsInfra = (
   opts: ProductDetailOptions,
 ) => Promise<{ errors: unknown[]; product: Product | null }>;
+
+export type GetProductBaseInfra = (
+  opts: ProductDetailOptions,
+) => Promise<{ errors: unknown[]; product: ProductBase | null }>;
```

Comment:
You should already be using Result type for this type and the types below:
```suggestion
export type GetProductBaseInfra = (
  opts: ProductDetailOptions,
) => AsyncResult<ProductBase | null>;
```